src/orchestratord/src/controller/materialize/environmentd.rs

```diff
@@ -73,6 +92,20 @@ pub enum DeploymentStatus {
     IsLeader,
 }
 
+#[derive(Deserialize, Serialize)]
+pub struct LoginCredentials {
+    username: String,
+    // TODO Password type?
+    password: ByteString,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ConnectionInfo {
```

Comment:
would it make sense to skip serializing the environmentd_url and mz_system_secret fields here? feels like the configmap contents are the only things we actually want to compare to see if there are changes to apply, and it might make things a bit easier to understand. probably not actually meaningfully different in practice though outside of debugging stuff.